### PR TITLE
QNAP NAS build

### PR DESCRIPTION
**Pull request by mattw-mega**
Scripts and instructions added for building MEGAcmd for QNAP NAS devices, and constructing a package that can be installed on the device.  Tested building for platform arm-x41, and running on a TS-228  so far.    That platform is 32-bit, and needed some small code changes to support that (and -fsigned-char for now on ARM).  Also checking those changes on a windows build, it turns out SOCKET is a different size for 32 and 64 bit, which I've addressed also, and various other warnings where the sizes of variables and parameters mismatched on a 32 or 64 bit build.

### CODE REVIEW

File: src/megacmdshell/megacmdshellcommunications.cpp [hunk 1]
```diff
@@ -264,11 +264,11 @@ bool is_pid_running(pid_t pid) {
 }
 #endif
 
-int MegaCmdShellCommunications::createSocket(int number, bool initializeserver, bool net)
+SOCKET MegaCmdShellCommunications::createSocket(int number, bool initializeserver, bool net)
 {
     if (net)
     {
-        int thesock = socket(AF_INET, SOCK_STREAM, 0);
+        SOCKET thesock = socket(AF_INET, SOCK_STREAM, 0);
```
Comment (polmr) [inline src/megacmdshell/megacmdshellcommunications.cpp]: int -> SOCKET should also be applied in the else of `if (net)`

File: build/QNAP_NAS/megacmdpkg/qpkg.cfg [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Name of the packaged application.
+QPKG_NAME="MEGAcmd"
+# Name of the display application.
+QPKG_DISPLAY_NAME="MEGAcmd"
+# Version of the packaged application. 
+QPKG_VER="0.9.8"
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply build/QNAP_NAS/megacmdpkg/qpkg.cfg]: yes, and that simplifies the instructions to copy files over to the NAS for the package build too, nice thought - done, thanks.

File: build/QNAP_NAS/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# Building a QNAP package
+
+This must be done on a QNAP NAS as it requires their QDK package.  Download QDK_2.2.16.qpkg and use the QNAP package manager to manually install that.
+
+Next, use `scp` or a similar tool to copy the MEGAcmd/build/QNAP_NAS/megacmdpkg folder to the NAS, followed by the contents of the MEGAcmd/install folder:
+`scp -r build/QNAP_NAS/megacmdpkg admin@10.12.0.248:/share/CACHEDEV1_DATA/.qpkg/QDK/`
+`scp -r install/* admin@10.12.0.248:/share/CACHEDEV1_DATA/.qpkg/QDK/megacmdpkg/arm-x41`
+
+On the NAS, using ssh or similar, edit the qpkg.cfg to adjust version numbers etc and then run the 'qbuild' command from the megacmdpkg folder.   The package will be built and put in the 'bulid' subfolder.
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply build/QNAP_NAS/README.md]: fixed, good spotting

File: build/QNAP_NAS/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# How to build MEGAcmd for QNAP NAS drives.  
+
+We follow the cross compiling system provided by QNAP, described at (https://www.qnap.com/event/dev/en/p_qdk.php#qdk_btn1_show) 
+
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply build/QNAP_NAS/README.md]: I've added the section, good thought

File: src/megacmdshell/megacmdshellcommunications.cpp (hunk 1 above)
Comment (mattw-mega) [reply src/megacmdshell/megacmdshellcommunications.cpp]: applied; well spotted

File: build/QNAP_NAS/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# How to build MEGAcmd for QNAP NAS drives.
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply build/QNAP_NAS/README.md]: added a small section in the main README that says which NAS systems we currently have scripts for, and where to find them.  cheers

**Pull request merged**